instrumentation/aws-sdk/aws-sdk-1.11/javaagent/build.gradle.kts

```diff
@@ -26,79 +26,91 @@ testSets {
 
-  testImplementation project(':instrumentation:aws-sdk:aws-sdk-1.11:testing')
+  testImplementation(project(":instrumentation:aws-sdk:aws-sdk-1.11:testing"))
 
-  testSqsImplementation "com.amazonaws:aws-java-sdk-sqs:1.11.106"
+  add("testSqsImplementation", "com.amazonaws:aws-java-sdk-sqs:1.11.106")
 
   // Include httpclient instrumentation for testing because it is a dependency for aws-sdk.
-  testInstrumentation project(':instrumentation:apache-httpclient:apache-httpclient-4.0:javaagent')
+  testInstrumentation(project(":instrumentation:apache-httpclient:apache-httpclient-4.0:javaagent"))
 
   // needed for kinesis:
-  testImplementation "com.fasterxml.jackson.dataformat:jackson-dataformat-cbor"
+  testImplementation("com.fasterxml.jackson.dataformat:jackson-dataformat-cbor")
 
   // needed for SNS
-  testImplementation "org.testcontainers:localstack:${versions["org.testcontainers"]}"
+  testImplementation("org.testcontainers:localstack:${versions["org.testcontainers"]}")
 
   // needed by S3
-  testImplementation 'javax.xml.bind:jaxb-api:2.3.1'
-
-  test_before_1_11_106Implementation("com.amazonaws:aws-java-sdk-s3:1.11.0")
-  test_before_1_11_106Implementation("com.amazonaws:aws-java-sdk-rds:1.11.0")
-  test_before_1_11_106Implementation("com.amazonaws:aws-java-sdk-ec2:1.11.0")
-  test_before_1_11_106Implementation("com.amazonaws:aws-java-sdk-kinesis:1.11.0")
-  test_before_1_11_106Implementation("com.amazonaws:aws-java-sdk-dynamodb:1.11.0")
-  test_before_1_11_106Implementation("com.amazonaws:aws-java-sdk-sns:1.11.0")
+  testImplementation("javax.xml.bind:jaxb-api:2.3.1")
+
+  add("test_before_1_11_106Implementation", "com.amazonaws:aws-java-sdk-s3:1.11.0")
+  add("test_before_1_11_106Implementation", "com.amazonaws:aws-java-sdk-rds:1.11.0")
+  add("test_before_1_11_106Implementation", "com.amazonaws:aws-java-sdk-ec2:1.11.0")
+  add("test_before_1_11_106Implementation", "com.amazonaws:aws-java-sdk-kinesis:1.11.0")
+  add("test_before_1_11_106Implementation", "com.amazonaws:aws-java-sdk-dynamodb:1.11.0")
+  add("test_before_1_11_106Implementation", "com.amazonaws:aws-java-sdk-sns:1.11.0")
 }
 
-test {
-  systemProperty "testLatestDeps", testLatestDeps
-}
+tasks {
+  val test_before_1_11_106 by existing(Test::class) {
+    filter {
+      // this is needed because "test.dependsOn test_before_1_11_106", and so without this,
+      // running a single test in the default test set will fail
+      setFailOnNoMatchingTests(false)
+    }
+  }
+
+  val testSqs by existing
 
-if (!testLatestDeps) {
-  check.dependsOn test_before_1_11_106, testSqs
-}
 
-tasks.withType(Test).configureEach {
-  // TODO run tests both with and without experimental span attributes
-  jvmArgs "-Dotel.instrumentation.aws-sdk.experimental-span-attributes=true"
+  if (!(findProperty("testLatestDeps") as Boolean)) {
+    named("check") {
+      dependsOn(test_before_1_11_106)
+      dependsOn(testSqs)
+    }
+  }
+
+  named<Test>("test") {
```

Comment:
Actually `test.configure {` at the top level is equivalent to this and arguably less verbose, but I figured consistency with other tests, which don't get such DSL, is simpler